src/corefx/System.Globalization.Native/localeStringData.cpp

```diff
@@ -240,3 +255,83 @@ extern "C" int32_t GetLocaleInfoString(const UChar* localeName, LocaleStringData
 
 	return UErrorCodeToBool(status);
 }
+
+/*
+Function:
+NormalizeTimePattern
+
+Convert an ICU non-localized time pattern to .NET format
+*/
+void NormalizeTimePattern(const UnicodeString *srcPattern, UnicodeString *destPattern)
```

Comment:
I was planning to have a similar function for Date patterns, but my plan was to write it in managed code.  What are you thoughts on this?  That way the native side stays really simple, and the "rule"/"logic" is in managed, i.e. string manipulation is easier/better.